letsencrypt/views/base_config_settings.xml

```diff
@@ -0,0 +1,27 @@
+<odoo>
+
+    <record id="base_config_settings" model="ir.ui.view">
+        <field name="name">Letsencrypt base config settings</field>
+        <field name="model">base.config.settings</field>
+        <field name="inherit_id" ref="base_setup.view_general_configuration" />
+        <field name="arch" type="xml">
+            <xpath expr="./group[last()]" position="after">
+                <group name="letsencrypt">
+                    <group name="letsencrypt" string="LetsEncrypt">
+                        <field name="letsencrypt_needs_dns_provider" />
+                        <field
+                            name="letsencrypt_shell"
+                            attrs="{'invisible': [('letsencrypt_needs_dns_provider', '!=', 'shell')]}" />
```

Comment:
add a placeholder text telling the user which environment variables are available

Reply:
What do you mean? Which environment variables?

Reply:
@hbrunn Ah just read your comment above, ignore

Reply:
A little bit lost here, where should the `_letsencrypt_shell_support` be called from? Aren't the environment variables defined by the user? By default there are no environment variables on the calling of this function.

Reply:
that's why I ask you to set them in https://github.com/OCA/server-tools/pull/1194/files#diff-9b7e43b14857ae1b9294fe0125a6846fR219

`_letsencrypt_shell_support` will be called in https://github.com/OCA/server-tools/pull/1194/files#diff-9b7e43b14857ae1b9294fe0125a6846fR195 as all others, so you need to keep the interface stable there. What I ask above is to make the function useful (how is a shell script now supposed to know for which domain to create which challenge?) by doing something like

```python
def shell_support(self, challenge, domain):
    ...
    self._call_cmdline(
        $your_commandline,
        ...,
        env=dict(os.environ, CHALLENGE=challenge, DOMAIN=domain),
    )
```

In general when programming something, place yourself in the shoes of the person using your program, then in this case you'll see you need some way to pass the information to the script.